app.js

```diff
@@ -23,21 +33,248 @@ app.use(express.static(path.join(__dirname, 'public')));
 app.set('view engine', 'jade');
 app.use(cors());
 
+// Session related plugins
+var session = require('express-session');
+var MongoStore = require('connect-mongo')(session);
+
 // Database
 var mongoUri = process.env.MONGOLAB_URI || process.env.MONGOHQ_URL|| 'mongodb://localhost/tusk';
 mongoose.connect(mongoUri);
 
+app.use(session({
+    secret: 'tuskislovetuskislife',
+    cookie: {
+        expires: Consts.genDefaultExpires(),
+        maxAge: Consts.getDefaultMaxAge()
+    },
+    saveUninitialized: false,
+    resave: true,
+    rolling: true,
+    unset: 'destroy',
+    store: new MongoStore({
+        mongooseConnection: mongoose.connection
+    })
+}));
+
+app.get('/user/:id/confirm', function(request, response) {
+    response.set('Content-Type', 'application/json');
+    Unconf_User.findOne({_id: request.params.id}, function(err, user) {
+        if (err) {
+            return response.send(JSON.stringify({
+                success: false,
+                type: 'INVALID_USER_ID',
+                message: 'Invalid user id'
+            }));
+        } else {
+            if (request.query.key === user.confirmationKey) {
```

Comment:
Do we want to have a password and a confirm password sent to the backend? Seems like we should check that this kind of thing on the front end. Just thinking out loud mainly so let me know if theres a reason we want to do that step on the backend